userspace/libsinsp/test/container_engine/container_parser_cri_containerd.ut.cpp

```diff
@@ -0,0 +1,749 @@
+		}
+	}
+	// CRI mounts
+	auto res = cri_api_v1alpha2->parse_cri_mounts(resp_container, container);
+	ASSERT_TRUE(res);
+	// CRI image
+	res = cri_api_v1alpha2->parse_cri_image(resp_container, resp_container_info, container);
+	ASSERT_TRUE(res);
+	ASSERT_EQ("3ad7b26ded6d8e7b23da7d48fe889434573036c27ae5a74837233de441c3601e", container.m_full_id);
+	ASSERT_EQ("docker.io/library/busybox:latest", container.m_image);
+	ASSERT_EQ("docker.io/library/busybox", container.m_imagerepo);
+	ASSERT_EQ("latest", container.m_imagetag);
+
+	// CRI image, failure resilience test
+	auto status = resp.mutable_status();
+	status->set_image_ref("sha256:3fbc632167424a6d997e74f52b878d7cc478225cffac6bc977eedfe51c7f4e79");
+	status->mutable_image()->set_image("");
+	const auto &resp_container_simulate_image_recovery = resp.status();
+	res = cri_api_v1alpha2->parse_cri_image(resp_container_simulate_image_recovery, resp_container_info, container);
+	ASSERT_TRUE(res);
+	ASSERT_EQ("docker.io/library/busybox:latest", container.m_image);
+	ASSERT_EQ("docker.io/library/busybox", container.m_imagerepo);
+	ASSERT_EQ("latest", container.m_imagetag);
+
+	// network
+	container.m_container_ip = ntohl(cri_api_v1alpha2->get_pod_sandbox_ip(resp_pod));
+	ASSERT_EQ(183762946, container.m_container_ip); // decimal of "10.244.0.2"
+	std::string cniresult = "";
+	cri_api_v1alpha2->get_pod_info_cniresult(resp_pod, cniresult);
+	ASSERT_EQ("{\"bridge\":{\"IPConfigs\":null},\"eth0\":{\"IPConfigs\":[{\"Gateway\":\"10.244.0.1\",\"IP\":\"10.244.0.2\"}]}}", cniresult);
+	container.m_pod_cniresult = cniresult;
+
+	// Extra info such as privileged flag
+	const auto &info_it = resp.info().find("info");
+	Json::Value root;
+	Json::Reader reader;
+	if(!reader.parse(info_it->second, root))
+	{
+		ASSERT(false);
+	}
+	res = cri_api_v1alpha2->parse_cri_ext_container_info(root, container);
+	ASSERT_TRUE(res);
+	ASSERT_TRUE(container.m_privileged);
+	ASSERT_EQ(1073741824, container.m_memory_limit);
+	ASSERT_EQ(50000, container.m_cpu_quota);
+
+	res = cri_api_v1alpha2->parse_cri_json_image(root, container);
+	ASSERT_TRUE(res);
+	ASSERT_EQ("busybox", container.m_imageid); // info.config.image.image can sometimes be in the format sha256: ...
+
+	//
+	// create and test sinsp_container_info for sandbox_container
+	//
+
+	// Add basic fields manually
+	sandbox_container.m_is_pod_sandbox = true;
+	sandbox_container.m_type = CT_CONTAINERD;
+	sandbox_container.m_id = "63060edc2d3a"; // truncated id extracted from cgroups, here for the sandbox id / pod
+	sandbox_container.m_full_id = resp_sandbox_container.id();
+	sandbox_container.m_name = resp_sandbox_container.metadata().name();
+	for(const auto &pair : resp_sandbox_container.labels())
+	{
+		if(pair.second.length() <= sinsp_container_info::m_container_label_max_length)
+		{
+			sandbox_container.m_labels[pair.first] = pair.second;
+		}
+	}
+
+	// 
+	// Test sinsp filterchecks, similar to spawn_process_container test
```

Comment:
I like this test but it is quite huge, maybe here in the sinsp filter-check part we could simplify a little bit things using just a `generate_clone_x_event` to create a child of init in the new container and assert all filter checks on it, WDYT? Follow all these execve/clone with all the 20/27 params is quite diffcult

Reply:
tried to push more empty or dummy params to make it look less crowded.